app/Hooks/SavePost.php

```diff
@@ -58,7 +58,7 @@ function post_sync_times_from_promo($post_id){
    */
   public function  promo_sync_times_to_posts($post_id, $post){
 
-    if ($post->post_type != 'promo'){
+    if(!$post || !isset($post->post_type) || $post->post_type !== 'promo'){
```

Comment:
I haven't bee doing this, but thoughtlet:

is it easier to wrap everything in `if ($post->post_type === 'promo') {} else return;`